File: ai/typescript/detect-anthropic.ts
// ruleid: detect-anthropic
import Anthropic from '@anthropic-ai/sdk';

// ruleid: detect-anthropic
const anthropic = new Anthropic({
  apiKey: 'my_api_key', // defaults to process.env["ANTHROPIC_API_KEY"]
});

// ruleid: detect-anthropic
const msg = await anthropic.messages.create({
  model: "claude-3-opus-20240229",
  max_tokens: 1024,
  messages: [{ role: "user", content: "Hello, Claude" }],
});
console.log(msg);

File: ai/python/detect-anthropic.py
# ruleid: detect-anthropic
import anthropic

# ruleid: detect-anthropic
client = anthropic.Anthropic(
    # defaults to os.environ.get("ANTHROPIC_API_KEY")
    api_key="my_api_key",
)

# ruleid: detect-anthropic
message = client.messages.create(
    model="claude-3-opus-20240229",
    max_tokens=1024,
    messages=[
        {"role": "user", "content": "Hello, Claude"}
    ]
)
print(message.content)